### PR TITLE
fix(IEC): move iec resource to services/iec and fix lint error

### PR DESCRIPTION
**Pull request by corningli**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
./scripts/codecheck.sh ./huaweicloud/services/iec

==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                       18      3673      448        32     3193        439           233.14
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~rvices/iec/resource_huaweicloud_iec_vip.go       313       42         9      262         55            20.99
~source_huaweicloud_iec_network_acl_rule.go       323       38         0      285         54            18.95
~rvices/iec/resource_huaweicloud_iec_eip.go       313       38         3      272         53            19.49
~ces/iec/resource_huaweicloud_iec_server.go       593       61        11      521         50             9.60
~ec/resource_huaweicloud_iec_network_acl.go       222       27         2      193         29            15.03
~rce_huaweicloud_iec_security_group_rule.go       221       26         0      195         29            14.87
~iec/resource_huaweicloud_iec_vpc_subnet.go       256       34         3      219         28            12.79
~resource_huaweicloud_iec_security_group.go       219       26         0      193         28            14.51
~rvices/iec/resource_huaweicloud_iec_vpc.go       147       24         1      122         18            14.75
~a_source_huaweicloud_iec_security_group.go       141       13         2      126         15            11.90
~es/iec/resource_huaweicloud_iec_keypair.go       111       16         0       95         14            14.74
~/data_source_huaweicloud_iec_bandwidths.go       137       21         0      116         14            12.07
~data_source_huaweicloud_iec_vpc_subnets.go       137       16         1      120         11             9.17
~iec/data_source_huaweicloud_iec_flavors.go       126       12         0      114         10             8.77
~/iec/data_source_huaweicloud_iec_images.go       112       12         0      100         10            10.00
~ces/iec/data_source_huaweicloud_iec_vpc.go        99       16         0       83          8             9.64
~es/iec/data_source_huaweicloud_iec_eips.go       141       17         0      124          8             6.45
~iec/data_source_huaweicloud_iec_keypair.go        62        9         0       53          5             9.43
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                    18      3673      448        32     3193        439           233.14
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 98704 bytes, 0.099 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
10 iec dataSourceIECSecurityGroupRead huaweicloud/services/iec/data_source_huaweicloud_iec_security_group.go:84:1
9 iec updateIecVipAssociate huaweicloud/services/iec/resource_huaweicloud_iec_vip.go:258:1
8 iec resourceIecVIPCreate huaweicloud/services/iec/resource_huaweicloud_iec_vip.go:78:1
7 iec resourceIecServerCreate huaweicloud/services/iec/resource_huaweicloud_iec_server.go:333:1
7 iec waitForSecurityGroupDelete huaweicloud/services/iec/resource_huaweicloud_iec_security_group.go:190:1
7 iec resourceIecSecurityGroupV1Read huaweicloud/services/iec/resource_huaweicloud_iec_security_group.go:120:1
7 iec resourceNetworkACLRuleCreate huaweicloud/services/iec/resource_huaweicloud_iec_network_acl_rule.go:116:1
7 iec waitForIecEipStatus huaweicloud/services/iec/resource_huaweicloud_iec_eip.go:286:1
7 iec resourceIecEipCreate huaweicloud/services/iec/resource_huaweicloud_iec_eip.go:105:1
7 iec dataSourceIECVpcSubnetIdsRead huaweicloud/services/iec/data_source_huaweicloud_iec_vpc_subnets.go:82:1
Average: 3.85

==> Checking for golangci-lint...

==> Checking for Nolint directives...
./huaweicloud/services/iec/resource_huaweicloud_iec_vpc.go:138: // lintignore:R018

==> Checking for TF features in iec...
  -> the resource in resource_huaweicloud_iec_network_acl_rule.go should can be imported


==> Checking for misspell in iec...

==> Checking for code complexity in ./huaweicloud/services/acceptance/iec...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                       18      1970      254         0     1716        155           141.13
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~uaweicloud_iec_security_group_rule_test.go       156       21         0      135         25            18.52
~e/iec/resource_huaweicloud_iec_eip_test.go       109       18         0       91         16            17.58
~esource_huaweicloud_iec_vpc_subnet_test.go       147       20         0      127         16            12.60
~e/iec/resource_huaweicloud_iec_vpc_test.go       182       25         0      157         16            10.19
~e/iec/resource_huaweicloud_iec_vip_test.go       177       23         0      154         16            10.39
~ec/resource_huaweicloud_iec_server_test.go       168       15         0      153         16            10.46
~rce_huaweicloud_iec_security_group_test.go       104       16         0       88         16            18.18
~source_huaweicloud_iec_network_acl_test.go       169       20         0      149          6             4.03
~ec/data_source_huaweicloud_iec_vpc_test.go        82       12         0       70          6             8.57
~e_huaweicloud_iec_network_acl_rule_test.go       109       14         0       95          5             5.26
~ata_source_huaweicloud_iec_flavors_test.go        81       11         0       70          3             4.29
~source_huaweicloud_iec_vpc_subnets_test.go        99       10         0       89          3             3.37
~data_source_huaweicloud_iec_images_test.go        53        8         0       45          3             6.67
~_source_huaweicloud_iec_bandwidths_test.go        88       10         0       78          3             3.85
~c/data_source_huaweicloud_iec_eips_test.go        88       10         0       78          3             3.85
~c/resource_huaweicloud_iec_keypair_test.go        69        9         0       60          2             3.33
~rce_huaweicloud_iec_security_group_test.go        47        7         0       40          0             0.00
~ata_source_huaweicloud_iec_keypair_test.go        42        5         0       37          0             0.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                    18      1970      254         0     1716        155           141.13
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 56792 bytes, 0.057 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
6 iec testAccCheckIecVpcV1Exists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_vpc_test.go:114:1
6 iec testAccCheckIecVpcSubnetV1Exists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_vpc_subnet_test.go:77:1
6 iec testAccCheckIecVipExists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_vip_test.go:95:1
6 iec testAccCheckIecServerExists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go:56:1
6 iec testAccCheckIecSecurityGroupV1Exists huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_security_group_test.go:47:1
Average: 2.17

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/iec...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/iec...

==> Cleanup patch...

Check Completed!
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/iec' TESTARGS='-run TestAccIECVpcDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iec -v -run TestAccIECVpcDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccIECVpcDataSource_basic
=== PAUSE TestAccIECVpcDataSource_basic
=== CONT  TestAccIECVpcDataSource_basic
--- PASS: TestAccIECVpcDataSource_basic (36.88s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec       36.923s
make testacc TEST='./huaweicloud/services/acceptance/iec' TESTARGS='-run TestAccIECSubnetsDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iec -v -run TestAccIECSubnetsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccIECSubnetsDataSource_basic
=== PAUSE TestAccIECSubnetsDataSource_basic
=== CONT  TestAccIECSubnetsDataSource_basic
--- PASS: TestAccIECSubnetsDataSource_basic (99.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec       99.421s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_vpc_test.go [hunk 1]
```diff
@@ -38,26 +38,26 @@ func testAccDataSourceIECVpcCheck(n, rName string) resource.TestCheckFunc {
...
 		rs, ok := s.RootModule().Resources[n]
 		if !ok {
-			return fmtp.Errorf("root module has no resource called %s", n)
+			return fmt.Errorf("root module has no resource called %s", n)
 		}
 
 		vpcRs, ok := s.RootModule().Resources["huaweicloud_iec_vpc.test"]
 		if !ok {
-			return fmtp.Errorf("can't find huaweicloud_iec_vpc.test in state")
+			return fmt.Errorf("can't find huaweicloud_iec_vpc.test in state")
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_vpc_test.go]: don't include `huaweicloud` in fmt.Errotf or log.Printf

File: huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_vpc_test.go [hunk 2]
```diff
@@ -38,26 +38,26 @@ func testAccDataSourceIECVpcCheck(n, rName string) resource.TestCheckFunc {
...
+			return fmt.Errorf(
 				"id is %s; want %s",
 				attr["id"],
 				vpcRs.Primary.Attributes["id"],
 			)
 		}
 
 		if attr["name"] != rName {
-			return fmtp.Errorf("bad iec vpc name %s", attr["name"])
+			return fmt.Errorf("bad iec vpc name %s", attr["name"])
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_vpc_test.go]: the `iec` should be updated to **IEC**

File: huaweicloud/services/iec/data_source_huaweicloud_iec_vpc_subnets.go [hunk 3]
```diff
@@ -116,12 +119,15 @@ func dataSourceIECVpcSubnetIdsRead(d *schema.ResourceData, meta interface{}) err
...
+		return diag.Errorf("error saving IEC subnets: %s", err)
 	}
 
 	// set id
 	d.SetId(hashcode.Strings(allIDs))
-	d.Set("region", GetRegion(d, config))
 
-	return nil
+	mErr := multierror.Append(nil,
+		d.Set("region", cfg.GetRegion(d)),
```
Comment (ShiChangkuo) [inline huaweicloud/services/iec/data_source_huaweicloud_iec_vpc_subnets.go]: subnets should also wrapped in multierror

```suggestion
		d.Set("region", cfg.GetRegion(d)),
		d.Set("subnets", iecSubnets),
```

File: huaweicloud/services/iec/data_source_huaweicloud_iec_vpc.go [hunk 4]
```diff
@@ -47,43 +50,50 @@ func DataSourceIECVpc() *schema.Resource {
...
+		return diag.Errorf("error creating IEC client: %s", err)
 	}
 
 	listOpts := vpcs.ListOpts{
 		ID:   d.Get("id").(string),
 		Name: d.Get("name").(string),
 	}
 
-	logp.Printf("[DEBUG] query VPCs using given filter: %+v", listOpts)
+	log.Printf("[DEBUG] query VPCs using given filter: %+v", listOpts)
```
Comment (houpeng80) [inline huaweicloud/services/iec/data_source_huaweicloud_iec_vpc.go]:
```suggestion
	log.Printf("[DEBUG] Query VPCs using given filter: %+v", listOpts)
```
and fix the other resource

File: huaweicloud/services/iec/data_source_huaweicloud_iec_vpc.go [hunk 5]
```diff
@@ -47,43 +50,50 @@ func DataSourceIECVpc() *schema.Resource {
 	}
 }
 
-func dataSourceIECVpcRead(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	iecClient, err := config.IECV1Client(GetRegion(d, config))
+func dataSourceIECVpcRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	iecClient, err := cfg.IECV1Client(cfg.GetRegion(d))
```
Comment (houpeng80) [inline huaweicloud/services/iec/data_source_huaweicloud_iec_vpc.go]:
```suggestion
        region := cfg.GetRegion(d)
	iecClient, err := cfg.IECV1Client(region)
```

File: huaweicloud/services/iec/data_source_huaweicloud_iec_vpc.go [hunk 6]
```diff
@@ -47,43 +50,50 @@ func DataSourceIECVpc() *schema.Resource {
...
-	d.Set("mode", vpcInfo.Mode)
-	d.Set("subnet_num", vpcInfo.SubnetNum)
-	d.Set("region", GetRegion(d, config))
+
+	mErr := multierror.Append(nil,
+		d.Set("name", vpcInfo.Name),
+		d.Set("cidr", vpcInfo.Cidr),
+		d.Set("mode", vpcInfo.Mode),
+		d.Set("subnet_num", vpcInfo.SubnetNum),
+		d.Set("region", cfg.GetRegion(d)),
```
Comment (houpeng80) [inline huaweicloud/services/iec/data_source_huaweicloud_iec_vpc.go]:
```suggestion
		d.Set("region", region),
```

File: huaweicloud/services/iec/data_source_huaweicloud_iec_vpc.go [hunk 7]
```diff
@@ -47,43 +50,50 @@ func DataSourceIECVpc() *schema.Resource {
...
 	if total > 1 {
-		return fmtp.Errorf("Your query returned more than one result." +
-			" Please try a more specific search criteria")
+		return diag.Errorf("your query returned more than one result," +
+			" please try a more specific search criteria")
 	}
 
 	vpcInfo := allVpcs.Vpcs[0]
-	logp.Printf("[DEBUG] Retrieved IEC VPC %s: %+v", vpcInfo.ID, vpcInfo)
+	log.Printf("[DEBUG] retrieved IEC VPC %s: %+v", vpcInfo.ID, vpcInfo)
```
Comment (houpeng80) [inline huaweicloud/services/iec/data_source_huaweicloud_iec_vpc.go]: for log info, the first letter of the first world shoud be in upper case

**Pull request merged**